### DISCUSSION TITLE
Consult-omni and Emacs > Alfred and OS X Spotlight

### DISCUSSION THREAD

**Discussion by ncouture** [armindarvish/consult-omni#32] [General]
# This is not a bug; it is my personal feedback on this project 

As an avid Emacs user that is currently looking into how other Emacs users are configuring the latest AI service providers APIs, code completion, documentation at point, and organization capabilities and everything else in between that will turn a developer into a cyborg, which happens to be the main reasons I go through all the trouble of configure Emacs the way I want it to work to suit my workflows every few years, I found your project interesting but wanted to share at least my point of view on a main aspect of it.

## Your project is attractive from multiple points of view

It sounds like you're on the brink of something remarkable with consult-omni. The journey of innovation often begins with a simple realization: the tools we have aren't quite meeting our needs. And that's where the magic happens—when you decide to build something better. 

## Power of Integration

By integrating multiple sources of data through popular APIs, you're not just creating a tool; you're crafting an ecosystem. This isn't about piecing together functionalities; it's about weaving a tapestry of capabilities that work in harmony. Each API is a thread, and together, they form a fabric that is stronger, more versatile, and more powerful than any single component.

## Demonstrating Value

Your video presentations are more than just demonstrations; they're narratives. They tell the story of a problem, a solution, and the journey in between. When you show how consult-omni solves issues that other solutions couldn't, you're not just selling a product—you're offering a new way of thinking, a new approach to tackling challenges.

## The Shift from Alfred and Spotlight

Choosing consult-omni over established tools like Alfred and OS X's Spotlight is a testament to its value. It's a bold statement that says, "This is not just another tool; this is the tool." It speaks volumes about the effectiveness and efficiency of what you've created. It's not just about replacing; it's about enhancing, about finding something that aligns more closely with your needs and aspirations.

## The Path Forward

As you continue to develop and refine consult-omni, remember that innovation is a journey, not a destination. Keep listening to your users, keep iterating, and keep pushing the boundaries of what's possible. The world is full of problems waiting for solutions, and with consult-omni, you're well on your way to making a significant impact.

In the end, it's not just about the tool itself, but about the change it brings, the problems it solves, and the new possibilities it opens up. Keep sharing your story, keep demonstrating your value, and watch as your creation becomes an indispensable part of the toolkit for many.

### On June 17 this year consult-omni received its first commit

One hundred twenty-one days later, your project has hundred forty-one stars on this project. This is more than a star a day, and I will be adding one as well.

### Collaboration

Additionally, if you need any assistance, especially with testing on Linux or integrating Docker for testing on multiple operating systems, I’d be happy to help. I have experience in these areas and am eager to contribute to the project’s success. Feel free to reach out to me at consult-omni@stormvault.net or @ncouture. Looking forward to the possibility of collaborating!

**Comment by armindarvish** [comment]
@ncouture First of all, thank you very much for the kind words. Yes, I do realize that consult-omni has great potential, and of course there are reasons beyond just making some simple tools for my use-case behind why I made consult-omni to begin with. I wrote a little bit about in this [blog post](https://www.armindarvish.com/post/web_omni_search_in_emacs_with_consult-web/).

I would love to get some more collaborators on this project. Testing in other platforms would be very useful. Also, I have not seen much of feedback on the package, its design or usage yet, so there is not much to change yet, but if you start testing it and come up with suggestions, I would appreciate that.

Other than that, I still need some help with the elsip code to improve the sorting and ordering of candidates, which has been covered in these discussions:
- https://github.com/armindarvish/consult-omni/discussions/1 
- https://github.com/armindarvish/consult-web/discussions/29

@karthink was looking into that at some point but I am guessing he is busy with his latex preview magic right now :D! I myself haven't had a chance to revisit that topic either because I have been busy getting my [consult-gh](https://github.com/armindarvish/consult-gh) into MELPA and also adding a whole lot of new functionalities to it.

That is then another daunting task right now, getting consult-omni into MELPA. As it is the package is designed to be modular and each source gets its own `.el` file with its own dependencies, etc. which is not exactly how MELPA recipes are set up. I don't know if it is worth to put this on MELPA since `package-vc` and `straight.el` are available now, but it may help reach more audience in the Emacs world!

Anyway, thanks for the kind words again and would love to collaborate with you. Just let me know how you want to proceed with collaborating.

**Reply by karthink** [reply]
@armindarvish Sorry about that, time for working on FOSS is limited right now and I'm busy with gptel bugs/the latex preview patchset.  I I will eventually sit down with the consult-omni code at some point and see how I can help.

**Comment by armindarvish** [comment]
@karthink not at all. I have not had much time to spend on consult-omni myself either. There is only so much time any of us can spend on FOSS projects.
